### PR TITLE
 React homework week3

### PR DESCRIPTION
**Pull request by islam-fawzy25**

### CODE REVIEW

**Comment by islam-fawzy25** [comment]
i found that it missing two requirements: Border component and more propTypes i will do that as soon as possible

**Review by calm-prog — COMMENTED** [review]
It looks quite good. Nicely structured, you paid attention to the details and really thought about how to write your code.

Some points you could have a look at though:

- Keep all your components separately. There is too much going on in your App.js. You can extract the components into separate files and import them there.
- When you use the map function it needs to return something (other than just calling a function).
- Make sure you don't modify states without using the setState() method.

File: react/week3/class/src/App.js [hunk 1]
```diff
@@ -0,0 +1,272 @@
...
+      })}
+    </div>
+  );
+};
+// TodoDescription component will execute new todo task by input tag and render AddTodo
+const TodoDescription = ({ todos, setTodos, inputValue, setInputValue }) => {
+  const inputFunction = (e) => {
+    setInputValue((todoValue) => {
+      if (e.target.value === "") {
+        return alert("write something here ");
```
[the remark this answers is not in the document]
Comment (islam-fawzy25) [reply react/week3/class/src/App.js:64]: i found out that i made a condition in addTodo fun line 114 so I shortcut my code here

**Comment by islam-fawzy25** [comment]
Thank you so much, MR.Calm for your Calmness, Patience on my code  and for those very valuable comments

**Pull request merged**